src/osd/OpRequest.h

```diff
@@ -152,13 +153,13 @@ struct OpRequest : public TrackedOp {
   void mark_reached_pg() {
     mark_flag_point(flag_reached_pg, "reached_pg");
   }
-  void mark_delayed(const std::string& s) {
+  void mark_delayed(const char* s) {
```

Comment:
By contrast, mark_delayed is *only* called with string literals:

```
src/osd/PrimaryLogPG.cc:  op->mark_delayed("waiting for missing object");                                                                                                                                          
src/osd/PrimaryLogPG.cc:  op->mark_delayed("waiting for degraded object");
src/osd/PrimaryLogPG.cc:  op->mark_delayed("waiting for cache not full");
src/osd/PrimaryLogPG.cc:  op->mark_delayed("waiting for clean to repair");
src/osd/PrimaryLogPG.cc:  op->mark_delayed("waiting for blocked object");
src/osd/PrimaryLogPG.cc:  op->mark_delayed("waiting for readable");
src/osd/PrimaryLogPG.cc:  op->mark_delayed("waiting for readable");
src/osd/PrimaryLogPG.cc:          op->mark_delayed("waiting for scrub");
src/osd/PrimaryLogPG.cc:          op->mark_delayed("waiting for readable");
src/osd/PrimaryLogPG.cc:    op->mark_delayed("waiting_for_map not empty");
src/osd/PrimaryLogPG.cc:    op->mark_delayed("op must wait for map");
src/osd/PrimaryLogPG.cc:      op->mark_delayed("waiting for peered");
src/osd/PrimaryLogPG.cc:    op->mark_delayed("waiting for flush");
src/osd/PrimaryLogPG.cc:      op->mark_delayed("waiting for active");
src/osd/PrimaryLogPG.cc:      op->mark_delayed("waiting for scrub");
src/osd/PrimaryLogPG.cc:        op->mark_delayed("waiting for ondisk");
src/osd/PrimaryLogPG.cc:    op->mark_delayed("waiting for rw locks");
src/osd/PrimaryLogPG.cc:        op->mark_delayed("waiting for scrub");
src/osd/PrimaryLogPG.cc:      op->mark_delayed("waiting for scrub");
src/osd/PrimaryLogPG.cc:  op->mark_delayed("waiting for missing object");
```

This conversion is therefore safe, though you need to invoke mark_flag_point rather than mark_flag_point_string.